process_report/process_report.py

```diff
@@ -58,8 +58,16 @@ def load_old_pis(old_pi_file):
     return old_pi_dict
 
 
+def dump_old_pis(old_pi_file, old_pi_dict: dict):
+    with open(old_pi_file, "w") as f:
+        for pi, first_month in old_pi_dict.items():
+            f.write(f"{pi},{first_month}\n")
+
+
 def is_old_pi(old_pi_dict, pi, invoice_month):
-    if pi in old_pi_dict and old_pi_dict[pi] != invoice_month:
+    if pi in old_pi_dict and old_pi_dict[pi] > invoice_month:
+        print(f"Found future PI {pi} from {old_pi_dict[pi]} in {invoice_month} invoice")
```

Comment:
This print is not necessary and this function can be simplified into only one `if` statement by using `.get` on the dictionary and providing a default value of the current month.

Reply:
We don't need to know if a future PI is encountered?

Reply:
There's no such thing as a "future" PI. If when re-processing the invoices of the past, you encounter a PI whose first encountered invoice is later than the invoice you're processing, there's something wrong with the invoices or the PI file. If you want to be extra careful you can raise an exception here and just stop the script altogether with an error.

Reply:
I've made it raise an exception, and added a test case for it.